corehq/apps/app_manager/static/app_manager/js/details/screen.js

```diff
@@ -188,9 +199,45 @@ hqDefine("app_manager/js/details/screen", function () {
         self.enableTilePullDown.subscribe(function () {
             self.saveButton.fire('change');
         });
-        self.columns.subscribe(function () {
+        self.columns.subscribe(function (changes) {
             self.saveButton.fire('change');
-        });
+
+            // create events for column moves and fire a special event that allows us to update
+            // dependent UI elements (sort properties)
+            const events = changes
+                // remove the 2nd event for column moves
+                .filter(c => !(c.status === 'deleted' && c.moved !== undefined));
+
+            // there should only be one 'change' now.
+            const change = events[0];
+
+            // for "moved" and "deleted" we need to add events for all the other columns that have changed their index
+            let affectedColumns, move;  // 'move' is an index diff to calculate the previous index
+            if (change.moved !== undefined) {
+                const moveFrom = change.moved,
+                    movedTo = change.index;
+                if (movedTo > moveFrom) {
+                    move = +1;
+                    affectedColumns = self.columns.slice(moveFrom, movedTo);
```

Comment:
I did start out with a data model change that adds a name field but decided to go for this in the end to avoid the data model change. It does make this piece of code more complex so there's a tradeoff. I could be persuaded to revert back to that